app/models/supported_ssg.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'yaml'
+
+# rubocop:disable Metrics/BlockLength
+SupportedSsg = Struct.new(:id, :package, :version, :profiles,
+                          :os_major_version, :os_minor_version,
+                          keyword_init: true) do
+  class << self
+    def all
+      raw_supported['supported'].map do |rhel, values|
+        major, minor = os_version(rhel)
+
+        new(
+          id: rhel,
+          os_major_version: major,
+          os_minor_version: minor,
+          **map_attributes(values)
+        )
+      end
+    end
+
+    def revision
+      raw_supported['revision']
+    end
+
+    private
+
+    def map_attributes(values)
+      attrs = values.slice(*members.map(&:to_s))
+      attrs['profiles'] = attrs['profiles']&.keys
+      attrs.symbolize_keys
+    end
+
+    def os_version(rhel)
+      rhel.scan(/(\d+)\.(\d+)$/)[0]
+    end
+
+    def raw_supported
+      YAML.load_file(supported_file)
```

Comment:
Could this be cached to reduce I/O?

Reply:
I cannot think of a simple way to do it. This is a classmethod, which would cache it for the duration of the app. Would you suggest a better way?

The file is access once for a general data retrieval. The API makes another call to get the revision. We can return the revision along side the collection to reduce the access. I believe that two I/O reads of the YAML are acceptable.

Reply:
Are these two calls per request or per boot?

If it is per request it would make sense to cache it with `@raw_supported ||= ...`.

Reply:
Two calls per request.

As I've mentioned `@raw_supported ||=` would store it for the whole duration of the app, disallowing to replace supported SSGs in runtime (via configmap).

Reply:
Having it cached for the duration of the app should be fine. The versions won't change often.
Too many request could cause instances to lock up because of I/O issues.

Reply:
Do I/O issues happen often on static files in deployed application? 

I don't mind adding the memoization. It looked to me as premature optimization.

Reply:
How about `Rails.cache` with invalidation?

Reply:
@akofink What do you think about the memoization for the whole runtime? It should be good enough for production, as change of a configmap would require a restart/re-deployment.

Reply:
I think there might be a way to parse and cache this file with [ruby config](https://github.com/rubyconfig/config) `Settings`.

Reply:
I've added the simpler memoization. Not sure how the config would help here. The Rails.cache could, but it is configured differently on each environment (null cache by default in dev).